paimon-common/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -127,6 +127,18 @@ public class CoreOptions implements Serializable {
                                                     + "if there is no primary key, the full row will be used.")
                                     .build());
 
+    @Immutable
+    public static final ConfigOption<HashType> BUCKET_HASH_TYPE =
```

Comment:
It seems we should introduce a BUCKET_FUNCTION first? Have you researched other systems with similar requirements?

Reply:
It seems the bucket function's implementation is `hash % bucket_number` for example Hive bucket, So the difference in different system is the hash function.